gap/base/recognition.gi

```diff
@@ -451,7 +451,20 @@ InstallGlobalFunction( RecogniseGeneric,
     # Find a possible homomorphism (or recognise this group as leaf)
     allmethods := methoddb;
     if IsBound(knowledge.hints) and Length(knowledge.hints) > 0 then
-        allmethods := Concatenation(knowledge.hints,allmethods);
+        # HACK
+        tmp := [];
+        for hint in knowledge.hints do
+            if IsRecogMethod(hint.method) then
+                Add(tmp, rec(rank := hint.rank, method := hint.method));
+            else
+                Add(tmp,
+                    rec(rank := hint.rank,
+                        method := rec(method := hint.method, stamp := hint.stamp)));
+            fi;
+            allmethods := Concatenation(tmp, allmethods);
```

Comment:
Huh, aren't you adding methods several times this way? You keep adding to tmp, then prepend it to allmethods -- again and again.

So either move this `Concatenation` out of the loop, or else just get rid of `tmp` and replace `Add(tmp` above by `Add(allmethods` (twice) -- the order should not matter as we sort by rank below anyway. 

Well, it might matter if several methods have identical rank, *and* if we were using `StableSortBy`.... I think having methods with equal rank is an error here, though, and perhaps we should add a corresponding assertion at the end. Something like this:
```gap
Assert(0, Length(Set(allmethods, m->m.rank)) = Length(allmethods)); # verify no rank occurs more than once
```

Reply:
added a fix